src/linux/Packaging.Linux/layout.sh

```diff
@@ -76,13 +72,22 @@ fi
 
 # Publish core application executables
 echo "Publishing core application..."
-$DOTNET_ROOT/dotnet publish "$GCM_SRC" \
-	--configuration="$CONFIGURATION" \
-	--framework="$FRAMEWORK" \
-	--runtime="$RUNTIME" \
-	--self-contained \
-	-p:PublishSingleFile=true \
-	--output="$(make_absolute "$PAYLOAD")" || exit 1
+if [ -z "$RUNTIME" ]; then
+    $DOTNET_ROOT/dotnet publish "$GCM_SRC" \
+        --configuration="$CONFIGURATION" \
+        --framework="$FRAMEWORK" \
+        --self-contained \
+        -p:PublishSingleFile=true \
+        --output="$(make_absolute "$PAYLOAD")" || exit 1
+else
+    $DOTNET_ROOT/dotnet publish "$GCM_SRC" \
+        --configuration="$CONFIGURATION" \
+        --framework="$FRAMEWORK" \
+        --runtime="$RUNTIME" \
+        --self-contained \
+        -p:PublishSingleFile=true \
+        --output="$(make_absolute "$PAYLOAD")" || exit 1
+fi
```

Comment:
I think we should keep layout.sh always producing a runtime-specific output.

Reply:
it does produce a runtime specific output. if runtime is not passed dotnet publish automatically determines the appropriate runtime from the running system. Refer to the linked actions in the first message showing that.

Reply:
To add to that, it would be very difficult to correctly determine each appropriate runtime to use ourselves for every type of system combination when there are dozens. Letting dotnet handle that (because it knows what runtime it was built on itself) is the correct way https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/src/PortableRuntimeIdentifierGraph.json#L146-L304